### PR TITLE
Added `gpiolib` and `sonata_gpio_demo_image`

### PR DESCRIPTION
**Pull request by HU90m**
This will be used for the first structured lab of the Hackathon.

### CODE REVIEW

**Comment by HU90m** [comment]
There's still a little more I'll add to the `hackathon.md` .

File: hackathon.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+```sh
+xmake config --board=sonata --debug-scheduler=y \
+    --debug-locks=y --debug-cxxrt=y \
+    --debug-token_library=y --debug-allocator=y
+xmake build
+```
+
+You may need to run `rm -rf build/ .xmake/` to remove the debug messages from the configuration.
+
```
Comment (GregAC) [inline hackathon.md:71]:
```suggestion
## Fallback Firmware load

We have been experiencing some reliability issues with copying firmware UF2 files to the SONATA virtual USB drive. You may find repeating the copy is sufficient to get things working but if you are having trouble we have a temporary backup load feature in the RP2040 firmware that uses the RP2040 USB boot/flash loader.

You will need to create a new UF2 from your firmware UF2 with the script in [`scripts/rp2040_uf2/`](scripts/rp2040_uf2/) 

# Script must be executed from the directory it lives in
cd scripts/rp2040_uf2
# Creates a file build/cheriot/cheriot/release/sonata_simple_demo_rp2040.uf2
./make_rp2040_uf2.sh ../../build/cheriot/cheriot/release/sonata_simple_demo.uf2

To use the new UF2 restart your board in RP2040 USB mode (hold the 'RP2040 boot' button whilst power cycling the board). Copy the new UF2 (build/cheriot/cheriot/release/sonata_simple_demo_rp2040.uf2 in the example above) to the RP2040 drive and this will load the Sonata firmware
```

(The shell stuff needs to be enclosed in backticks but these break the github UI suggestion thing)

**Review by marnovandermaas — APPROVED** [review]
approving this so it can be merged when ready.

File: compartments/led_walk_raw.cc
Comment (nbdd0121) [inline compartments/led_walk_raw.cc:1]: Move this file in a separate commit to register as move.
